src/components/core/PivotTable.tsx

```diff
@@ -313,19 +316,25 @@ export class PivotTableInner extends BaseVisualization<IPivotTableInnerProps, IP
             if (this.isGrowToFitEnabled(prevProps) !== this.isGrowToFitEnabled()) {
                 this.growToFit(this.columnApi);
             }
-            const prevColumnWidths = this.getColumnWidths(prevProps);
-            const columnWidths = this.getColumnWidths(this.props);
-            if (!isEqual(prevColumnWidths, columnWidths) && !this.waitingForFirstExecution) {
-                if (this.columnApi) {
-                    this.resizedColumnsStore.updateColumnWidths(columnWidths, this.getExecutionResponse());
 
-                    syncSuppressSizeToFitOnColumns(this.resizedColumnsStore, this.columnApi);
-
-                    if (this.isGrowToFitEnabled()) {
-                        this.growToFit(this.columnApi); // calls resetColumnsWidthToDefault internally too
+            if (!isEqual(prevColumnWidths, columnWidths) && !this.columnWidthsChangeWaitingForExecution) {
+                if (this.columnApi) {
+                    if (this.shouldWaitForExecution(totalsPropsChanged, totalsStateChanged)) {
+                        this.columnWidthsChangeWaitingForExecution = true;
```

Comment:
Just wondering if it has some reason why `this.columnWidthsChangeWaitingForExecution = true;` is not assigned only at one place in the code. 
Please try how it would look if you put it into one condition. If you would end up with a lot of duplication of the if statements above, feel free to keep as is :-)

Reply:
just moved columnApi condition